src/libstd/collections/hash/table.rs

```diff
@@ -768,25 +770,38 @@ impl<K, V> RawTable<K, V> {
                                                            align_of::<HashUint>(),
                                                            pairs_size,
                                                            align_of::<(K, V)>());
-        assert!(!oflo, "capacity overflow");
+        if oflo {
+            return Err(CollectionAllocErr::CapacityOverflow);
+        }
 
         // One check for overflow that covers calculation and rounding of size.
-        let size_of_bucket = size_of::<HashUint>().checked_add(size_of::<(K, V)>()).unwrap();
-        assert!(size >=
-                capacity.checked_mul(size_of_bucket)
-                    .expect("capacity overflow"),
-                "capacity overflow");
+        let size_of_bucket = size_of::<HashUint>().checked_add(size_of::<(K, V)>())
```

Comment:
You turn a `unwrap()` into a return with `CollectionAllocErr::CapacityOverflow` here, which I think technically changes the behaviour (or at least the panic message). Is this just a bugfix because this is also technically a capacity overflow?

Reply:
In my understanding, if bucket allocation size cannot be calculated, it's a sign of `capacity overflow`.  I don't think there is a regression risk here because the chances of hitting this case are probably very small. There are 2 things that will change here (1) non try_xxxx methods will have a different panic message (empty vs "capacity overflow") (2) the behaviour might change when oom=panic/abort is introduced.

Reply:
Yeah, makes sense.